### ISSUE TITLE
Cannot capture contents of linkedin.com

### ISSUE THREAD

**Issue by pbernery** [wallabag/wallabag#2114]
### Issue details

I tried to capture this [site](https://www.linkedin.com/pulse/agile-dead-matthew-kern) and it did not work, creating a fake article with "wallabag can't retrieve contents for this article. Please report this issue to us.…" as content.
### Environment
- wallabag version 2.0.4
- Installed via `git clone`
- php version: 5.6.20-0+deb8u1
- OS: debian 8
- type of hosting (shared or dedicated): dedicated
- which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL):
### Steps to reproduce/test case
1. Copy and paste this URL https://www.linkedin.com/pulse/agile-dead-matthew-kern into the new entry field of wallabag.
2. Validate

**Comment by j0k3r** [comment]
_(developer note for later)_ Works fine on f43.me. Strange..

**Comment by Strubbl** [comment]
Works for me too. And also works with FTR: http://ftr.fivefilters.org/makefulltextfeed.php?url=https%3A%2F%2Fwww.linkedin.com%2Fpulse%2Fagile-dead-matthew-kern&max=3

**Comment by nicosomb** [comment]
It works for me locally and on my own webserver. But not on v2.wallabag.org (I really think we have a problem on this platform). 

@pbernery I close this issue. Please reopen if you still have the problem.

**Comment by HLFH** [comment]
@nicosomb See https://github.com/wallabag/wallabag/issues/1399#issuecomment-264329682

**Comment by JoeCotellese** [comment]
It seems that LinkedIn does stuff to prevent scraping. I tried setting the User Agent to something I knew worked locally. However, my deployed version on Digital Ocean returns 999 when a request is made to LinkedIn

Found this and believe it's related https://github.com/yatish27/linkedin-scraper/issues/44

So, how do we get Graby to use a proxy list?

**Comment by HLFH** [comment]
@nicosomb @j0k3r Couldn't we use Tor + Privoxy ? http://www.dsfc.net/logiciel-libre/configuration-privoxy-tor-scrapping/

Or as a workaround,  couldn't we be able to edit failed scraped posts and add to it some plaintext note as a replacement of a void post ?

**Comment by j0k3r** [comment]
At the moment, we can't use a proxy for graby.

**Comment by Suncatcher** [comment]
Linkedin rendering is still not working

![image](https://cloud.githubusercontent.com/assets/6388034/26262005/f3eb7248-3cdb-11e7-8290-c61de3f82a6d.png)

I can confirm OP article and cann add another example:
https://www.linkedin.com/pulse/employees-from-mars-entrepreneurs-venus-steve-faktor

Wallabag version: 2.2.2
Platform: AWS

**Comment by borisschapira** [comment]
LinkedIn Pulse is different from other LinkedIn source. Here is a proposal but I don't know why it does not work in https://f43.me/feed/test (I've tried to respect the syntax but may have missed s.t.):

```
title: //h1[@class=‘article-title’]
body: //div[@class='prose’]
author: //span[@itemprop=‘name’]
date: //time

test_url: https://www.linkedin.com/pulse/mixit-2017-jai-fait-le-plein-de-cr%C3%AApes-et-damour-emilie-esposito
```

**Comment by JoeCotellese** [comment]
@borisschapira your solution doesn't get around the problem that LinkedIn blocks access to their content based on IP addresses. 

If I deploy Wallabag locally on my home computer everythign works fine. If I deploy on DigitalOcean I get an error.

You can verify whether or not your IP address is blocked just by curling the URL from your system.

**Comment by borisschapira** [comment]
Oh, I did not know that LinkedIn was doing that kind of thing. Have you tried to call the URL with a falsy Google User-Agent (maybe they have open the content for Google)? :p

**Comment by JoeCotellese** [comment]
Yup. I changed the User agent. If I recall, it's both the user agent and the
IP. It was originally failing for me locally. I then changed the UA string and
it worked. When I deployed it to Digital Ocean it stopped working again. Digging
further I uncovered the IP address block.
You'd have to fix this with a proxy but I don't  believe there are plans for
that based on what I read in the thread above.
Joe  





On Mon, Jun 12, 2017 10:47 AM, Boris SCHAPIRA notifications@github.com  wrote:
Oh, I did not know that LinkedIn was doing that kind of thing. Have you tried to
call the URL with a falsy Google User-Agent (maybe they have open the content
for Google)? :p
